### PR TITLE
Switch CI to ansible-2.3.1.0

### PR DESCRIPTION
**Pull request by sdodson**
This only gets tox and friends but I'll rev ansible on the CI slave too.

### CODE REVIEW

**Comment by sdodson** [comment]
[test]

**Comment by sosiouxme** [comment]
Actually the origin build with playbook2image also uses the `requirements.txt`. However that's changing.

Main concern is integration tests, checking those locally along with the other tox stuff.

File: tox.ini [hunk 1]
```diff
@@ -24,9 +24,6 @@ commands =
     generate_validation: python setup.py generate_validation
     # TODO(rhcarvalho): check syntax of other important entrypoint playbooks
     ansible_syntax: python setup.py ansible_syntax
-    # ansible 2.2.2+ unfortunately breaks the integration test runner
-    # because it can no longer set facts on the test docker hosts.
-    # So for now, install separate ansible version for integration.
     # PR that fixes it: https://github.com/ansible/ansible/pull/23599
     # Once that PR is available, drop this and use same ansible.
-    integration: pip install ansible==2.2.1.0
+    integration: pip install ansible==2.3.1.0
```
Comment (sosiouxme) [inline tox.ini]: no real reason to keep this or the comment above. could replace with:

`    integration: python -c 'print "run test/integration/run-tests.sh"'`

Comment (sdodson) [inline tox.ini]: updated

**Comment by sdodson** [comment]
aos-ci-test

**Comment by sdodson** [comment]
[merge][severity:blocker]

**Comment by sdodson** [comment]
I'm considering this a blocker because ansible-2.3.1.0 will release in RHEL 7.4 extras and we need to ensure that we pivot to that version ahead of that.

**Pull request merged**